RecoMuon/MuonRechitClusterProducer/plugins/RechitClusterProducer.cc

```diff
@@ -0,0 +1,268 @@
+#include <memory>
+#include <cmath>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/MuonReco/interface/MuonRecHitCluster.h"
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/ClusterSequence.hh"
+#include "fastjet/PseudoJet.hh"
+
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/PtEtaPhiMass.h"
+#include "DataFormats/DTRecHit/interface/DTRecHitCollection.h"
+#include "DataFormats/CSCRecHit/interface/CSCRecHit2DCollection.h"
+#include "DataFormats/MuonReco/interface/MuonRecHitCluster.h"
+
+typedef std::vector<reco::MuonRecHitCluster> RecHitClusterCollection;
+
+template <typename Trait>
+class RechitClusterProducerT : public edm::global::EDProducer<> {
+  typedef typename Trait::RecHitRef RechitRef;
+  typedef typename Trait::RecHitRefVector RecHitRefVector;
+
+public:
+  explicit RechitClusterProducerT(const edm::ParameterSet&);
+  ~RechitClusterProducerT() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  const edm::ESGetToken<typename Trait::GeometryType, MuonGeometryRecord> GeometryToken_;
+  edm::EDGetTokenT<typename Trait::InputType> InputToken_;
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.
```suggestion
  const edm::ESGetToken<typename Trait::GeometryType, MuonGeometryRecord> geometryToken_;
  edm::EDGetTokenT<typename Trait::InputType> inputToken_;
```